WalkingChallenge/Challenge/Challenge.swift

```diff
@@ -31,4 +31,161 @@ import UIKit
+    }
+
+    btnSettings.setImage(UIImage(named: "GearIcon"), for: .normal)
+
+    view.addSubview(btnSettings)
+    btnSettings.snp.makeConstraints {
+      $0.top.equalTo(topLayoutGuide.snp.bottom)
+      $0.right.right.equalToSuperview().inset(Style.Padding.p16)
+    }
+
+    tblCards.allowsSelection = false
+    tblCards.backgroundColor = #colorLiteral(red: 0.9647058823, green: 0.9725490196, blue: 0.9803921568, alpha: 1.0000000000)
+    tblCards.dataSource = self
+    tblCards.delegate = self
+    tblCards.separatorStyle = .none
+    tblCards.estimatedRowHeight = 44
+    tblCards.rowHeight = UITableView.automaticDimension
+
+
+    tblCards.estimatedRowHeight = 44
+    tblCards.rowHeight = UITableView.automaticDimension
+
+    tblCards.register(JourneyMilesCardView.self)
+    tblCards.register(MilestoneDetailCardView.self)
+
+    tblCards.register(JourneyCardView.self)
+
+    view.addSubview(tblCards)
+    tblCards.snp.makeConstraints {
+      $0.top.equalTo(lblTitle.snp.bottom).offset(Style.Padding.p16)
+      $0.bottom.equalToSuperview().inset(Style.Padding.p16)
+      $0.left.right.equalToSuperview().inset(Style.Padding.p16)
+    }
+
+    arrCards.append(JourneyCardMiles())
+
+    tblCards.reloadData()
+  }
+
+  private func getTitle(_ title: String, subTitle: String) -> NSMutableAttributedString {
+
+    let paragraphStyle = NSMutableParagraphStyle()
+    paragraphStyle.lineSpacing = 4
+
+    let titleFont = [NSAttributedString.Key.font: Style.Typography.headerTitle.font!, .paragraphStyle: paragraphStyle]
+    let subFont = [NSAttributedString.Key.font: Style.Typography.smallRegular.font!, .paragraphStyle: paragraphStyle]
+    let titleAttr = NSMutableAttributedString(string: title, attributes: titleFont)
+    let subTitleAttr = NSMutableAttributedString(string: "\(subTitle)", attributes: subFont)
+
+    let combination = NSMutableAttributedString()
+    combination.append(titleAttr)
+    combination.append(subTitleAttr)
+    return combination
+  }
+}
+
+extension Challenge: UITableViewDataSource {
+
+  func numberOfSections(in tableView: UITableView) -> Int {
+    return arrCards.count
+  }
+
+  func tableView(_ tableView: UITableView,
+                 numberOfRowsInSection section: Int) -> Int {
+    if arrCards[section].expand == true {
+      return arrCards[section].miles.count + 1
+    }
+    return 1
+  }
+
+  func tableView(_ tableView: UITableView,
+                 cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    guard let card = arrCards[safe: indexPath.section] else {
+      return UITableViewCell()
+    }
+
+    if indexPath.row == 0 {
+      let cell: UITableViewCell =
+        tableView.dequeueReusableCell(withIdentifier: card.renderer,
+                                      for: indexPath)
+      guard let view = cell as? JourneyMilesCardView else { return UITableViewCell() }
+      view.render(card)
+
+      view.showMileStone = {
+        self.arrCards[indexPath.section].expandMile(true)
```

Comment:
Why is `expandMile` not a property?  If there is something that needs to occur on the set why not use an observable?  The `didSet` should allow you to post-process the change.